nodejs/week3/meal_sharing_website/src/backend/api/meals.js

```diff
@@ -0,0 +1,184 @@
+      if (error) {
+        console.error(`error ${error}`);
+      } else if (results.length === 0) {
+        response.send('No match found');
+      }
+      mealsFunctionParams(request, response);
+    });
+  }
+}
+
+// This function is for DELETE operation
+const mealsFunctionDelete = function (request, response) {
+  const idString = request.params.id;
+  const id = parseInt(idString);
+  if (isNaN(id)) {
+    response.send('Id should be an integer');
+  } else {
+    pool.query('DELETE FROM meal  WHERE id = ?', id, function (error, results) {
+      if (error) {
+        console.error(`error ${error}`);
+      } else if (results.affectedRows === 0) {
+        response.send('No match found');
+      }
+      response.send({ Deleted: results.affectedRows > 0 });
+    });
+  }
+}
+
+// This function to pass specific queries based on the query param
+function getMealsDBquery(object) {
+  if (isEmpty(object)) {
+    return `SELECT * FROM meal;`
+  } else if (object.maxPrice) {
+    return `SELECT * FROM meal WHERE price < ${object.maxPrice};`
+  } else if (object.availableReservations) {
+    return `select * from meal where max_reservations > (select number_of_guests from reservation where reservation.meal_id = meal.id);`
+  } else if (object.title) {
+    return `SELECT * FROM meal WHERE title LIKE '%${object.title}%';`
+  } else if (object.createdAfter) {
+    return `SELECT * FROM meal WHERE created_date > ${object.createdAfter};`
+  } else if (object.limit) {
+    return `SELECT * FROM meal LIMIT ${object.limit};`
+  }
+}
+
+// This function is to do some validations for the query params
+function isValidQueryParam(object) {
+  if (isEmpty(object)) {
+    return null;
+  } else if (object.maxPrice) {
+    const maxPriceInInt = parseInt(object.maxPrice);
+    if (isNaN(maxPriceInInt)) {
+      return 'Maximum Price should be an integer';
+    } else {
+      return null;
+    }
+  } else if (object.availableReservations) {
+    if (typeof(object.availableReservations) !== 'boolean') {
+      return 'AvailableReservations should be a boolean';
+    } else {
+      return null;
+    }
+  } else if (object.title) {
+    return null;
+  } else if (object.createdAfter) {
+    const dateInString = object.createdAfter;
+    if (!moment(dateInString, 'YYYY-MM-DD', true).isValid()) {
+      return 'Date should always be a valid date';
+    } else {
+      return null;
+    }
+  } else if (object.limit) {
+    const limitInt = parseInt(object.limit);
+    if (isNaN(limitInt)) {
+      return 'Limit should be an integer';
+    } else {
+      return null;
+    }
+  }
+}
+// Function to check if the array has a match to the parameters in the endpoint
+function processArrayLength(array, res) {
+  if (array.length === 0) {
+    res.send('No match found');
+  } else {
+    res.json(array);
+  }
+}
+
+// This function is to check if request.params or request.query object is empty
+function isEmpty(obj) {
```

Comment:
This check can be done in a shorter way:

```
return Object.keys(obj).length === 0
```

Unless you're concerned about the support in different environments.